### PR TITLE
faster forward following what is done for images

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
Follow up of #21897

Keeps batch_size active, but restricts it to the imagepart.
Calculates candidate_labels features only once (there's no way with this current pipeline to send different candidate_labels so we can take the optimization.
Ultra narrowed for CLIP but the pipeline has existed for 4 months now with no new models. Given the popularity of CLIP for diffusion models, I think this is ok to overspecify. We can always relax later when new models come in.
This allows to downgrade from ChunkPipeline to regular PIpeline

### CODE REVIEW

File: src/transformers/pipelines/zero_shot_audio_classification.py [hunk 1]
```diff
@@ -1,26 +1,22 @@
+from collections import UserDict
```
Comment (sgugger) [inline src/transformers/pipelines/zero_shot_audio_classification.py:1]: Unrelated but this is missing copyright.

**Review by sgugger — APPROVED** [review]
LGTM, but let's wait for @Narsil to have a look as well.

**Review by Narsil — APPROVED** [review]
LGTM Thank you !

**Comment by Narsil** [comment]
Failure seems unrelated

**Pull request merged**